src/coreclr/vm/prestub.cpp

```diff
@@ -1034,23 +1014,6 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
     COR_ILMETHOD_DECODER ilDecoderTemp;
     COR_ILMETHOD_DECODER *pilHeader = GetAndVerifyILHeader(pConfig, &ilDecoderTemp);
     *pFlags = pConfig->GetJitCompilationFlags();
-
-#ifdef FEATURE_TIERED_COMPILATION
-    bool isTier0 = pFlags->IsSet(CORJIT_FLAGS::CORJIT_FLAG_TIER0);
-    // If we've already opted-out of call counting, for example in the UnmangedCallersOnly case,
-    // switch to optimized code.
-    if (!shouldCountCalls)
-    {
-        pFlags->Clear(CORJIT_FLAGS::CORJIT_FLAG_TIER0);
-        pFlags->Clear(CORJIT_FLAGS::CORJIT_FLAG_TIER1);
```

Comment:
Have you validated that the JIT correctly generates optimized code in this case (and not Tier-0 code)?

Reply:
Yea `TieredCompilation::GetJitFlags` sends the correct JIT flags now when tiering is disabled